File: SGT.WebAdmin/ViewsScripts/Enumeradores/EnumTipoConsolidacao.js
var EnumTipoConsolidacaoHelper = function () {
    this.NaoConsolida = 0;
    this.AutorizacaoEmissao = 1;
    this.PreCheckIn = 2;
}

EnumTipoConsolidacaoHelper.prototype = {
    obterOpcoes: function () {
        return [
            { text: "Não Consolida", value: this.NaoConsolida },
            { text: "Autorização Emissão", value: this.AutorizacaoEmissao },
            { text: "Pré CheckIn", value: this.PreCheckIn }
        ];
    },
}

var EnumTipoConsolidacao = Object.freeze(new EnumTipoConsolidacaoHelper());

File: Dominio/ObjetosDeValor/Embarcador/Enumeradores/EnumTipoConsolidacao.cs
namespace Dominio.ObjetosDeValor.Embarcador.Enumeradores
{
    public enum EnumTipoConsolidacao
    {
        NaoConsolida = 0,
        AutorizacaoEmissao = 1,
        PreCheckIn = 2,
    }

    public static class EnumTipoConsolidacaoHelper
    {
        public static string ObterDescricao(this EnumTipoConsolidacao tipoConsolidacao)
        {
            switch (tipoConsolidacao)
            {
                case EnumTipoConsolidacao.NaoConsolida: return "Não Consolida";
                case EnumTipoConsolidacao.AutorizacaoEmissao: return "Autorizaçã Emissão";
                case EnumTipoConsolidacao.PreCheckIn: return "Pré CheckIn";
                default: return "Não Consolida";
            }
        }

    }
}
